enhancements/proposals/node-autoreplacement.md

```diff
@@ -0,0 +1,133 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: <podName>-autoreplacement-config
+  ...
+data:
+  autoreplacement.config: |
+    replace-address-first-boot: <clusterIP>    
+```
+
+The config is later merged with generic config during Scylla setup, which effectively triggers running Scylla with `replace-address-first-boot` flag.
+
+It is necessary to check whether a node is first in the cluster, as the first node in the cluster skips bootstrap, and therefore the flag can't be set.
+Checking for the IP already existing in gossip is necessary as it is not possible to replace a node that does not.
+Setting `replace-address-first-boot` results in a desired output, i.e. a node which lost its storage will be replaced during bootstrap.
+Otherwise, setting the flag is safe as Scylla ignores it after bootstrap.
+
+Introducing this enhancement additionally allows for removing replacement functionality from `OrphanedPVController`. The controller would be simplified and only be responsible for removing the orphaned PVs.
+
+### User Stories
+
+#### Story 1
+
+The user is running a ScyllaCluster and looses a PVC bound to one of Scylla nodes. The node gets automatically replaced. 
+
+#### Story 2
+
+The user is running a ScyllaCluster on AWS/GCP cluster with a local storage. One of the instances is subject to an event resulting in erasing the local storage (see [AWS Instance store lifetime](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html#instance-store-lifetime) and [GCP Local SSD data persistance](https://cloud.google.com/compute/docs/disks/local-ssd#data_persistence))  .
+A node scheduled on this instance loses its storage, even though a PVC bound to it is not removed. The node gets replaced automatically.
+
+### Notes/Constraints/Caveats [Optional]
+
+- It would be preferred to use [Immutable ConfigMaps](https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-immutable).
+- Having introduced this feature, we might consider removing the existing functionality of [replacing dead nodes manually](https://operator.docs.scylladb.com/master/nodeoperations/replace_node.html).
+
+### Risks and Mitigations
+
+- The proposed design is based on a fact that `replace-address-first-boot` flag is ignored after bootstrap. Any changes to this behaviour or side effects introduced could result in an undefined behaviour of the enhancement.
+
+## Design Details
+
+To check whether the node is not the first to bootstrap in a cluster and already exists in gossip, the controller uses [`ScyllaClient's Status`](https://github.com/scylladb/scylla-operator/blob/689c0a0a82aeea0fef72b2cd11af0333d1d2e180/pkg/scyllaclient/client.go#L98).
+Client is created with default config. Controller authenticates with the already known token. It communicates with all nodes of a given cluster. The conditions for creating the `ConfigMap` are met if the node's IP is among the returned hosts.
+
+`Sidecar` accesses the `ConfigMap` through a dedicated `Volume`.
+
```

Comment:
details I am missing mentioned here:
- is the configmap optional or required
- is the CM live reloaded
- will any of it result in spec update / rollout
- possible races
- in which cases the controller talks to the nodes, what is the condition to not ask anymore
- how is the CM (holding the IP) updated when a node is replaced manually (with the replace label)
- what happens when a manual replace is already in progress
- what happens if a user config contains the replace-address-first-boot option (not that we should support it but the behaviour should be defined - like our takes precedence)

Reply:
I don't think I saw a mention of precedence with manual replace using the label